Code/Framework/AzGameFramework/AzGameFramework/Application/GameApplication.cpp

```diff
@@ -101,11 +111,34 @@ namespace AzGameFramework
         AzFramework::Application::CreateStaticModules(outModules);
 
         outModules.emplace_back(aznew AzGameFrameworkModule());
+#if !LY_HEADLESS
+        outModules.emplace_back(aznew AzFramework::AzFrameworkNativeUIModule());
+#endif // LY_HEADLESS
     }
 
     void GameApplication::QueryApplicationType(AZ::ApplicationTypeQuery& appType) const
     {
         appType.m_maskValue = AZ::ApplicationTypeQuery::Masks::Game;
+        if (m_headless)
+        {
+            appType.m_maskValue |= AZ::ApplicationTypeQuery::Masks::Headless;
+        }
     };
 
+    AZ::ComponentTypeList GameApplication::GetRequiredSystemComponents() const
+    {
+        AZ::ComponentTypeList components = AzFramework::Application::GetRequiredSystemComponents();
+
+        #if !LY_HEADLESS
+        components.insert(
+            components.end(),
+            {
+                azrtti_typeid<AzFramework::NativeUISystemComponent>(),
+            });
+        #endif // LY_HEADLESS
```

Comment:
Ok, this answers my earlier question, about the `AzFrameworkNativeUiModule` not overriding the GetRequiredComponents function to add the `NativeUiSystemComponent` as a required system component.

That being said, since there is a block up above in this file to add the AzFrameworkNativeUIModule behind the `O3DE_HEADLESS_SERVER` define, then I believe the GetRequiredSystemComponents function SHOULD be overridden in `AzFrameworkNativeUIModule` module and the `GameApplication` shouldn't need to override the `Application::GetRequiredSystemComponents` function at all.

This logic isn't needed, because if the AzFrameworkNativeUiModule is not added, it would not add its required system components.
If the AzFrameworkNativeUiModule is added, then it should its `GetRequiredSystemComponents` function will be called in that case.